### PR TITLE
Escape LLVM_TARGETS_TO_BUILD while checking against LLVM_ALL_TARGETS and LLVM_EXPERIMENTAL_TARGETS_TO_BUILD

### PR DESCRIPTION
**Pull request by natschz**
Targeting X86 while building LLVM with the Android NDK, currently fails with "The target `X86' is not a core tier target. It may be experimental, if so it must be passed via LLVM_EXPERIMENTAL_TARGETS_TO_BUILD."

This is happening because the Android NDK defines a variable named X86, which will lead to a "fun" CMake feature. For further details reference [this post](https://discourse.llvm.org/t/the-target-x86-is-not-a-core-tier-target/73784).

To fix this, the LLVM_TARGETS_TO_BUILD need to be escaped while checking agains LLVM_ALL_TARGETS and LLVM_EXPERIMENTAL_TARGETS_TO_BUILD.

### CODE REVIEW

File: llvm/CMakeLists.txt [hunk 1]
```diff
@@ -942,7 +942,7 @@ foreach(t ${LLVM_TARGETS_TO_BUILD})
   # LLVM_EXPERIMENTAL_TARGETS_TO_BUILD, not LLVM_TARGETS_TO_BUILD.
   # We allow experimental targets that are not in LLVM_ALL_EXPERIMENTAL_TARGETS,
   # as long as they are passed via LLVM_EXPERIMENTAL_TARGETS_TO_BUILD.
-  if ( NOT ${t} IN_LIST LLVM_ALL_TARGETS AND NOT ${t} IN_LIST LLVM_EXPERIMENTAL_TARGETS_TO_BUILD )
+  if ( NOT "${t}" IN_LIST LLVM_ALL_TARGETS AND NOT "${t}" IN_LIST LLVM_EXPERIMENTAL_TARGETS_TO_BUILD )
     if( ${t} IN_LIST LLVM_ALL_EXPERIMENTAL_TARGETS )
```
Comment (smeenai) [inline llvm/CMakeLists.txt:946]: Might as well fix this one too, while you're here?

**Review by smeenai — APPROVED** [review]
LGTM, thanks! Will you need someone to land this for you?

File: llvm/CMakeLists.txt (hunk 1 above)
Comment (smeenai) [reply llvm/CMakeLists.txt:946]:
```suggestion
    if( "${t}" IN_LIST LLVM_ALL_EXPERIMENTAL_TARGETS )
```

**Comment by natschz** [comment]
> LGTM, thanks! Will you need someone to land this for you?

Yea I think so, not quite sure what to do next.

**Pull request merged**